src/vivarium_cluster_tools/vipin/log_parser.py

```diff
@@ -10,82 +9,68 @@
 import requests
 
 
-class WorkerLog:
-
+class PerformanceLog:
     def __init__(self, file: Path):
         self.file = file
-        name = file.stem.split('.')  # TODO: validate name is <sge job id>.<task id>.log
-        self.sge_job_id = int(name[0])
-        self.task_id = int(name[1])
+        self.count = 0
 
     def get_summaries(self) -> dict:
-        """Get all performance summary log messages in WorkerLog"""
-        telemetry_pattern = re.compile(r'^\{\"host\".+\"job_number\".+\}$')
+        """Get all performance summary log messages in PerformanceLog"""
+        telemetry_pattern = re.compile(r'^{\"host\".+\"job_number\".+}$')
 
-        # Ideally we'd only need to look at the tail of the file, but because a worker can do multiple draws, we need
-        # to iterate over all the lines...
-        # TODO: either change worker to cut a new log for each draw and check tail reversed(list(log))
-        #  *OR* parallelize this
+        self.count = 0
         f = self.file.open('r')
         for line in f.readlines():
             message = json.loads(line)['record']['message']
             m = telemetry_pattern.fullmatch(str(message))
             if m:
+                self.count += 1
                 yield json_normalize(json.loads(message), sep='_')
         f.close()
 
 
-def parse_log_directory(input_directory: Union[Path, str], output_directory: Union[Path, str], output_hdf: bool,
-                        random_draw: int = 0):
+def parse_log_directory(input_directory: Union[Path, str], output_directory: Union[Path, str], output_hdf: bool):
     input_directory, output_directory = Path(input_directory), Path(output_directory)
-    worker_log_pattern = re.compile(r'^([0-9]+)\.([0-9]+)\.log$')
-    log_files = [f for f in input_directory.iterdir() if worker_log_pattern.fullmatch(f.name)]
-
-    if random_draw:
-        log_files = sample(log_files, random_draw)
+    perf_log = PerformanceLog(Path(input_directory / 'performance.log'))
 
-    worker_data = []
-    for i, file in enumerate(log_files):
-        # TODO: set up tdqm progress bar
-        logger.info(f'Parsing file {i+1} of {len(log_files)}.')
-        log = WorkerLog(file)
-        for item in log.get_summaries():
-            worker_data.append(item)
-    worker_df = pd.concat(worker_data)
+    perf_data = []
```

Comment:
Why not a `perf_log.to_df()` method?

Reply:
That's a good suggestion. I'll change that.